pkg/ddl/partition.go

```diff
@@ -4229,6 +4251,9 @@ func hexIfNonPrint(s string) string {
 }
 
 func writeColumnListToBuffer(partitionInfo *model.PartitionInfo, sqlMode mysql.SQLMode, buf *bytes.Buffer) {
+	if partitionInfo.IsEmptyColumns {
+		return
+	}
```

Comment:
I think it would be OK to only see the PARTITION BY KEY() as syntactic sugar, i.e. always expand it to PARTITION BY KEY (<list of PK columns>), including in the SHOW CREATE TABLE output. But this is of course more MySQL compatible.